app/services/agency_identity_linker.rb

```diff
@@ -9,6 +9,20 @@ def link_identity
       AgencyIdentity.new(user_id: @sp_identity.user_id, uuid: @sp_identity.uuid)
   end
 
+  def self.for(user:, service_provider:)
```

Comment:
Since technically this can return either an AgencyIdentity or a ServiceProviderIdentity, I'd drop a little YARD comment with types as a hint
```suggestion
  # @return [AgencyIdentity, ServiceProviderIdentity] the AgencyIdentity for this user at this service provider
  #   or falls back to the ServiceProviderIdentity if one does not exist 
  def self.for(user:, service_provider:)
```